plugins/cloudbuild/src/components/WorkflowRunDetails/WorkflowRunDetails.tsx

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Entity } from '@backstage/catalog-model';
+import { Link, WarningPanel } from '@backstage/core';
+import {
+  Breadcrumbs,
+  LinearProgress,
+  Link as MaterialLink,
+  makeStyles,
+  Paper,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableRow,
+  Theme,
+  Typography,
+} from '@material-ui/core';
+import ExternalLinkIcon from '@material-ui/icons/Launch';
+import qs from 'qs';
+import React from 'react';
+import { useProjectName } from '../useProjectName';
+import { WorkflowRunStatus } from '../WorkflowRunStatus';
+import { useWorkflowRunsDetails } from './useWorkflowRunsDetails';
+
+const useStyles = makeStyles<Theme>(theme => ({
+  root: {
+    maxWidth: 720,
+    margin: theme.spacing(2),
+  },
+  title: {
+    padding: theme.spacing(1, 0, 2, 0),
+  },
+  table: {
+    padding: theme.spacing(1),
+  },
+  accordionDetails: {
+    padding: 0,
+  },
+  button: {
+    order: -1,
+    marginRight: 0,
+    marginLeft: '-20px',
+  },
+  externalLinkIcon: {
+    fontSize: 'inherit',
+    verticalAlign: 'bottom',
+  },
+}));
+
+export const WorkflowRunDetails = ({ entity }: { entity: Entity }) => {
+  const { value: projectName, loading, error } = useProjectName(entity);
+  const [projectId] = (projectName ?? '/').split('/');
+
+  const details = useWorkflowRunsDetails(projectId);
+
+  const classes = useStyles();
+  if (error) {
+    return (
+      <WarningPanel title="Error:">
+        Failed to load build, {error.message}.
+      </WarningPanel>
+    );
+  } else if (loading) {
+    return <LinearProgress />;
+  } else if (details.value?.logUrl === undefined) {
+    return <LinearProgress />;
```

Comment:
this is peculiar - does data arrive twice, once without this field and once with it?

Reply:
Yes, it happens that sometimes i get an undefined on the first request but on the next one it is defined. I am not entirely sure why... any help with this would be appreciated.